app/models/publication.rb

```diff
@@ -373,4 +298,76 @@ def authoritative_pmid_source?
   end
 
   alias authoritative_doi_source? sciencewire_pub?
+
+  private
+
+    def add_all_identifiers_in_db_to_pub_hash
+      publication_identifiers.reload if persisted?
+      db_ids = publication_identifiers.collect do |id|
+        ident_hash = {}
+        ident_hash[:type] = id.identifier_type if id.identifier_type.present?
+        ident_hash[:id] = id.identifier_value if id.identifier_value.present?
+        ident_hash[:url] = id.identifier_uri if id.identifier_uri.present?
+        ident_hash
+      end
+      pub_hash[:identifier] = db_ids
+    end
```

Comment:
PR #400 intersects this move-method refactor.  It would be good to see this move-method refactor rebased on #400 if/when we merge #400 on master.

Reply:
On master
```
git grep 'add_all_identifiers_in_db_to_pub_hash'
app/models/publication.rb:166:    add_all_identifiers_in_db_to_pub_hash
app/models/publication.rb:314:  def add_all_identifiers_in_db_to_pub_hash
lib/scripts/publication_identifier_normalization.rb:29:    pub.add_all_identifiers_in_db_to_pub_hash
```
In #384, the identifier normalization is refactored to avoid calling this method.  If we can work to get #400 and #384 merged, this move-method refactor could work.  This method is called in a save-callback, in an attempt to sync the pub-hash with the saved db-content (all these save-callback sync calls are a nightmare).